### ISSUE TITLE
Preview link is not correct in multi site scenario

### ISSUE THREAD

**Issue by rengert** [PiranhaCMS/piranha.core#335]
**How to reproduct**

1. setup

Create site 1 with url 1
Create site 2 with url 2

2. create page

Use url 2 and create a page below site 1

3. preview

Clicks preview button.

This redirects to slug on site 2 and ends in 404.

**How to fix**

Well, this depends on the target scenario and how the preview mode is implemented. 

One scenario could be the cms is running using a cms url, to make the cms not working using frontend url (security approach some are using). In this case the preview should run using the content management url and should be able to preview all sites.

Or is should be redirected to frontend with additional information, to ensure the preview is visible for the current user and only for the current user.

**Comment by tidyui** [comment]
Yes this is how Piranha has always worked. The preview mode opens the slug of the current page in a new tab on the **same domain** as the admin interface is running. This only works if:

* You access the manager UI from the domain you want to preview on.
* The manager interface is running in the same web application as the front-end.

Instead of using the current request domain of the admin we could use one of the hostname specified on the site, however hostnames are **not required** as most use cases are single sitemap scenarios. I'm open for ideas but the issue has always been that Piranha is not in any way aware of the front-end application rendering the pages, therefor it's hard to do anything other than send the manager to the public site.

**Comment by rengert** [comment]
Well, i do not think so. For now i have only to sites, but in the future more. 

Either the link:
- is linked correctly using same domain (and is working)
- is linked using the frontend domain (and is working you know, not published content and user not logged in)
- is not showing 404 but a more readable error message ("you may want to check the ...")
- cms is using domain hopping - maybe not a good aproach changing the domain as soon as you edit a page

For me the fix or change request is not so important anymore, i do not use it. Just want to let you know.

**Comment by tidyui** [comment]
The proposed fix to this is to use the **first** hostname entered in the site (if multiple hostnames have been bound to the site). If there's no hostname bound to the site then the preview will use the same hostname as the manager (current behavior).

**Comment by steelspace** [comment]
Just to mention that this behavior is not incorrect just for preview in Manager but also on site where the permalinks are missing the hostname part.

**Comment by tidyui** [comment]
Not sure what you mean @steelspace. The hostname is not supposed to be in the slug. The hostname is only used to determine the current site when a request comes into the middleware pipeline.

**Comment by tidyui** [comment]
Sorry for not responding sooner. Slugs **only** contains a unique string made up of the page's hierarchical path AND title, it does not contain any part of any hostname entered in the site. The reason for this is that you test your site on `localhost` and deploy it on `mysite.com`, adding the hostname would make even this simple scenario fail. When linking internally in a site all generated links are **relative**, so they use the current domain.

The hostnames are **only** used (when having multiple sites) to determine **which** site is requested when a HTTP request comes in. The problem with previewing pages from `Site B`, when you are logged into the manager on `Site A` is that generating a relative URL will fail. On the other hand, generating an **absolute** URL could generate a link to another domain where the admin is not logged in, which means he/she won't see the preview anyway.

This has always been the case in Piranha CMS. The simple solution is to actually log into `siteb.com/manager` when previewing content on that site, and log into `sitea.com/manager` when previewing content on that site.
